File: C programming/Assignment/Assignment9/strong.c
#include <stdio.h>

int isStrong(int num);
int fact(int rem);
void main(){
    int num;
    printf("Enter the number:");
    scanf("%d",&num);

    if(isStrong(num)==1)
        printf("Given number is strong");
    else
        printf("Given number is not strong");
}
int isStrong(int num){
    int temp = num;
    int sum = 0;
    while(num>0){
        int rem = num % 10;
        sum = sum + fact(rem);
        num = num / 10;
    }
    if(sum==temp)
        return 1;
    return 0;
}
int fact(int rem){
     int fact = 1;
     for(int i=1;i<=rem;i++){
        fact = fact * i;
     }
     return fact;
}

File: Java_programming/basic_java/Strong.java
package type4;

public class Strong {

	public static void main(String[] args) {
		int num=145;
//	    printf("enter the number:");
//	    scanf("%d",&num);
	    if(strong(num)==1)
	        System.out.printf("strong number");
	    else
	        System.out.printf("not strong number");

	}
	
	public static int strong(int num){
	    int fact,sum=0,res,mod;
	    
	    res = num;
	    while(num!=0){
	        mod = num % 10;
	        fact = 1;
	        while(mod!=0){
	            fact =fact * mod;
	            mod--;
	        }
	        sum = sum + fact;
	        num = num / 10;
	    }
	    if(res == sum)
	        return 1;
	    return 0;
	}

}
